### PR TITLE
Scripts/Mulgore: Fix Bugs Quest 14491

### PR DESCRIPTION
**Pull request by Nightprince**
* Fix exploit on SpellHit
* Add Talk when change faction to 14

### CODE REVIEW

File: src/server/scripts/Kalimdor/zone_mulgore.cpp [hunk 1]
```diff
@@ -77,20 +78,21 @@ class npc_agitated_earth_spirit : public CreatureScript
...
+                    {
+                        me->setFaction(35);
+                        if (Player* player = ObjectAccessor::GetPlayer(*me, playerGUID))
+                            player->KilledMonsterCredit(NPC_EARTH_SPIRIT_CREDIT_BUNNY);
+
+                        me->DespawnOrUnsummon(1000);
+                        break;
+                    }
+                    case 1:
+                        Talk(0);
```
Comment (AriDEV) [inline src/server/scripts/Kalimdor/zone_mulgore.cpp]: this wont work, the type of the message in db is 42 currently in skyfire it is (CHAT_MSG_RAID_BOSS_WHISPER) for that message you need to set the target in this case its player.

Talk(0, player);

File: sql/updates/world/2020_03_05_06_world.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+DELETE FROM `creature_text` WHERE `entry`=36845 AND `groupid`=0 AND `id`=0;
+INSERT INTO `creature_text` (`entry`, `text`, `type`, `probability`, `sound`, `comment`) VALUES (36845, 'The spirit is displeased and attacks!', 42, 100, 33398, 'Agitated Earth Spirit to Player');
```
Comment (AriDEV) [inline sql/updates/world/2020_03_05_06_world.sql]: the sound id is not correct.

Comment (AriDEV) [reply sql/updates/world/2020_03_05_06_world.sql]: crash

**Comment by AriDEV** [comment]
crash

**Comment by HBSNET** [comment]
It crashing the server!

**Comment by Nightprince** [comment]
> 
> 
> crash



> 
> 
> It crashing the server!

Crash not from my codes.

> 00007FF70309B700  0000006A777FF550  CreatureTextLocalizer<CreatureTextBuilder>::operator()+2F0  F:\SkyFire.548\src\server\game\Texts\CreatureTextMgr.h line 163
> 00007FF703096BBE  0000006A777FF5F0  CreatureTextMgr::SendChatPacket<CreatureTextBuilder>+8E  F:\SkyFire.548\src\server\game\Texts\CreatureTextMgr.h line 196
> 00007FF70309DAB6  0000006A777FF730  CreatureTextMgr::SendChat+6F6  F:\SkyFire.548\src\server\game\Texts\CreatureTextMgr.cpp line 282
> 00007FF70316ED97  0000006A777FF7A0  CreatureAI::Talk+57  F:\SkyFire.548\src\server\game\AI\CreatureAI.cpp line 45

**Comment by LichKing255** [comment]
> crash

> It crashing the server!

core have problem , not from my codes.




> It crashing the server!

this commit fix crash :

https://github.com/ProjectSkyfire/SkyFire_548/commit/3bb3fec39b138b09891dac670e213109f5b80e5c

**Comment by LichKing255** [comment]
this commit fix crash:
https://github.com/ProjectSkyfire/SkyFire_548/commit/3bb3fec39b138b09891dac670e213109f5b80e5c

**Pull request merged**